src/app/tests/TestBindingTable.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/util/binding-table.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+
+using chip::BindingTable;
+using chip::NullOptional;
+
+namespace {
+
+void TestEmptyBindingTable(nlTestSuite * aSuite, void * aContext)
```

Comment:
Thank you, this file makes me a lot more confident in the iterator implementation....